proposals/eni.md

```diff
@@ -0,0 +1,128 @@
+                                            |      Routing Table:    |   |
+                                            |   +------------------+ |   |
+                                          +----+|default via eth1  | +   |
+                             172.31.13.14 |eth1||169.254.170.2 via | |   |
+                                          +----+|     dev ve_en1   | +   |
+                                            |   +------------------+ |---+
+                                            |       169.254.x.2      +
+                                            +-----------+------------+
+                                                        |ve_en1
+                                                        |
+                                                        |
+                                                        |
+                                                        |
+                          +-------------------------------------------------------------+
+                          |default(global) namespace    |                               |
+                          |                             |                               |
+                          |                     +-------+-------+       +-------------+ |  +-----------+
+                          |                     |    ve_br_en1  |       |             | |  |net="bridge|
+             172.31.0.226 |   +-----------+     |               |       |             +-|--|container  |
+                       +----+ |           |     |ecs-eni bridge |       |             | |  +-----------+
+                       |eth0| | ECS Agent |     |               |       | docker0 bridge|
+                       +----+ +-----------+     |               |       |             | |
+                          |                     |               |       |             | |
+                          |                     |  169.254.x.1  |       |             | |
+                          |                     +---------------+       +-------------+ |
+                          |                                                             |
+                          +-------------------------------------------------------------+
+
+In this diagram, the primary network interface is still being used on the instance
+by the `docker0` bridge to let containers launched with `net=bridge` to communicate
+with entities outside the instance. A secondary network interface has also been
+attached for the purpose of being used by a task (`eth1`). This interface has been
+moved to the `ENI-1` network namespace. All containers of the task will share this
+network namespace.
+
+Tasks (containers in tasks) launched into these namespaces will be addressable
+by the primary IP Address of the respective secondary network interfaces. For
+example, all containers of the task using ENI `ENI-1` will be addressable within
+the VPC using the IP Address `172.31.13.14/20`.
```

Comment:
The `/20` here indicates the size of the subnet in which the ENI has been created. The IP Address is whatever's allocated by the VPC when the ENI was created. On, the host, it's still scoped to the `/20` address range so that traffic from this ENI can reach other network interfaces on the VPC using the gateway. 

>Do we need to do anything special to enable IPv6 here?

If the ENI has been configured with an IPv6 address, we probably should set that address on the interface as well. I haven't played with that yet, but it would be along the same lines as setting the IPv4 address.
@bchav Each container in a task doesn't get its own IP Address. All containers in the task share the network interface and hence are addressable by the IP Address of the ENI.